lib/roby/custom_require.rb

```diff
@@ -0,0 +1,26 @@
+# This is needed because of Ruby bug #9244
+module Kernel
+    alias syskit_original_require require
+    def require(path)
+        # Only deal with paths starting with models/, the other ones are
+        # unambiguous
+        if path =~ /^models\//
+            Roby.app.search_path.each do |app_dir|
+                full_path = File.join(app_dir, path)
+                if File.file?(full_path) || File.file?("#{full_path}.rb")
+                    return syskit_original_require(full_path)
+                end
+            end
+        elsif (match = /^(\w+)\/models\//.match(path))
```

Comment:
just to mention, it could be `/^(?<app_name>\w+)/models\//` so it automatically creates the app_name variable and you don't need to catch the 'match' return.

Reply:
Just looked it up ... it's a nice feature. However, it would not create a variable, "only" make the capture named instead of numbered (i.e. having `match[:app_name]` instead of `match[1]`. For such a simple regexp, I'm not entirely sure of the advantage (it does make the regexp harder to read IMO)... but I'm very used to them. What do you think ?

Reply:
Regexp are quite cryptographic to me sometimes, naming the capture just helps with finding were on the regex is the group (definitely not hard in this case, I get both).. On the `match[:app_name] ` thing, it is true when creating a match obj, but when using "=~" you do get a new local variable (and you don't need the match obj). In the end, it was just to mention, I believe it is more a matter of style.

Reply:
> but when using "=~" you do get a new local variable (and you don't need the match obj)

I'm actually trying to avoid `=~` with captures, the whole "use not-really global variables and/or magically created variables" has been getting on my nerves.

Thanks for the discussion anyways. I'd like to stick with the current code if that's all the same to you